javascript/ql/lib/semmle/javascript/security/performance/ReDoSUtil.qll

```diff
@@ -627,13 +676,27 @@ RegExpRoot getRoot(RegExpTerm term) {
   result = getRoot(term.getParent())
 }
 
-private newtype TState =
+/**
+ * A state in the NFA.
+ */
+newtype TState =
```

Comment:
Why make this public? (leftover from testing?)

Reply:
I need it to be public in the query I'm developing. 
I just copy-pasted the `ReDoSUtil.qll` file from my development of that query into this separate PR, so that's why I removed the `private` modifier.

Reply:
I've made them private again.